Assets/00_Content/Scripts/Interactables/Interactable.cs

```diff
@@ -1,16 +1,15 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-using UnityEngine;
+﻿using UnityEngine;
 
 namespace Interactables {
 	public class Interactable : MonoBehaviour {
 
 		public virtual void CancelInteraction() {
 		}
-		public virtual void Interact() {
+		public virtual void Interact(GameObject player, PickUp item) {
+		}
+
+		public virtual bool CanInteract(GameObject player, PickUp item) {
+			return true;
```

Comment:
I dont see a reason to force an interaction to require an item. I can see examples where we want to interact with things not using an item. I would just use a method overload for when an item is required.

Reply:
The idea is that the player always sends along whatever it is holding, and then let the interactable decide if such an interaction is possible. The interactable can choose to ignore the item if it doesn't care.

I think having an overload would just complicate the usage. You would have to implement both overloads if you want to cover both cases. I believe that just always passing along the held item makes the most sense.

Reply:
So we just send NULL when interacting with things not requireing an item or if we don´t hold an item.